src/wallet.rs

```diff
@@ -15,8 +15,8 @@ pub fn request_airdrop(
         airdrop_request_amount: tokens,
         client_pubkey: *id,
     };
-    let tx = serialize(&req).expect("serialize drone request");
-    stream.write_all(&tx).unwrap();
+    let tx = serialize(&req)?;
```

Comment:
This error case is pretty much impossible, right? If so, I wonder if changing that error message to IoError might further improve error messaging (or at least error handling).

Reply:
I don't think it changes it, you just get back whatever the library error reports. Actually I would like to be able to tell between these three and also get the original error as well, but I guess I need to make it custom, right?

Reply:
I confirmed `connect()` and `write_all()` both return a `std::io::Result`, so should be no need for that `Result<(), Box<error::Error>` if you make this line `serialize().unwrap()`, which I think is perfectly reasonable. I haven't tried to pattern match on a `Result<(), Box<error::Error>>`. You might be able to downcast it, but an experiment that doesn't need test-driving here. A function should probably only use the `Box<Error>` trick if you don't plan to handle the error, like in `main()` or a test.